pkg/filters/args.go

```diff
@@ -55,10 +58,21 @@ func (filter *ArgFilter) Filter(eventID events.ID, args []trace.Argument) bool {
 		if !found {
 			return false
 		}
-		// TODO: use type assertion instead of string conversion
-		if argName != "syscall" {
-			argVal = fmt.Sprint(argVal)
+
+		argVal = fmt.Sprint(argVal)
+		if argName == "syscall" {
+			// TODO: this only handles args set as syscall name, not syscall id.
+			// To handle syscall id, changes are needed in the parser to store
+			// an unique identifier (or syscall name or the syscall id).
+			syscallID, err := strconv.Atoi(argVal.(string))
```

Comment:
The type conversion is unnecessary as you're already working with a string.

Reply:
the assign above doesn't change the argVal, so the assertion is required.